File: webfront/web/models/product/product.js
define(function(require) {
    //requirements
    var Model = require('kit/model/model'),
        normalizeNumber = require('kit/normalizeNumber/normalizeNumber'),
        _ = require('lodash');

    return Model.extend({
        urlRoot: Model.baseApiUrl + '/products',
        defaults: {
            vat: 0,
            units: 'шт'
        },
        saveData: function() {

            var model = this,
                purchasePrice = normalizeNumber(model.get('purchasePrice')),
                sellingPrice = normalizeNumber(model.get('sellingPrice'));

            return {
                subCategory: model.get('subCategory'),
                name: model.get('name'),
                units: model.get('units'),
                barcode: model.get('barcode'),
                vat: model.get('vat'),
                purchasePrice: _.isNaN(purchasePrice) ? model.get('purchasePrice') : purchasePrice,
                sellingPrice: _.isNaN(sellingPrice) ? model.get('sellingPrice') : sellingPrice,
                type: 'unit'
            }
        }
    });
});

File: dreamkas-api-client/src/main/java/ru/dreamkas/api/objects/Product.java
package ru.dreamkas.api.objects;

import org.json.JSONException;
import org.json.JSONObject;
import ru.dreamkas.api.objects.abstraction.AbstractObject;

public class Product extends AbstractObject {

    private static final String API_URL = "/products";

    public Product(JSONObject jsonObject) {
        this.jsonObject = jsonObject;
    }

    public Product(String name,
                   String units,
                   String barcode,
                   String vat,
                   String purchasePrice,
                   String sellingPrice,
                   String groupId) throws JSONException {
        this(new JSONObject()
                        .put("type", "unit")
                        .put("name", name)
                        .put("units", units)
                        .put("barcode", barcode)
                        .put("vat", vat)
                        .put("purchasePrice", purchasePrice)
                        .put("sellingPrice", sellingPrice)
                        .put("subCategory", groupId)
        );
    }

    @Override
    public String getApiUrl() {
        return API_URL;
    }

    public String getSku() throws AssertionError {
        return getPropertyAsString("sku");
    }

    public String getBarCode() throws AssertionError {
        return getPropertyAsString("barcode");
    }
}
